### PR TITLE
Migrate CI to GH Actions

### PR DESCRIPTION
**Pull request by okken**
* Migrate CI to GitHub Actions
* Officially Support Python 3.7+ 
* drop testing of Python 2.7, 3.5, 3.6
* add testing of Python 3.10, 3.11

### CODE REVIEW

**Comment by okken** [comment]
@nicoddemus Is there a PR review and release process for pytest-dev plugins?

**Comment by nicoddemus** [comment]
@okken I don't think there is, but if you are looking for a template to follow, I suggest [pytest-xdist](https://github.com/pytest-dev/pytest-xdist) or [pytest-mock](https://github.com/pytest-dev/pytest-mock).

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,2 @@
-[bdist_wheel]
-universal = 1
+;[bdist_wheel]
```
Comment (nicoddemus) [inline setup.cfg:1]: We can delete this, we do not support Python 2 anymore.

**Review by nicoddemus — APPROVED** [review]
LGTM!

I would have thought GH Actions would already kick in for the PR...

File: setup.cfg (hunk 1 above)
Comment (okken) [reply setup.cfg:1]: cool. I'll delete the file.

**Comment by okken** [comment]
@nicoddemus Thanks for your help. 
I think it's worth doing more small pushes to PyPI, rather than wait until lots of changes are made.
Right now, having the README updated might be enough to reflect the versions of Python that are supported and tested, and update the banner icons.

So, there are options. 
- Either build locally and publish with twine.
- Integrate a publish workflow in GH actions.

Do I need to ask @davehunt for me to be added as a maintainer on PyPI?

**Comment by nicoddemus** [comment]
> I think it's worth doing more small pushes to PyPI, rather than wait until lots of changes are made.
Right now, having the README updated might be enough to reflect the versions of Python that are supported and tested, and update the banner icons.

Agreed.

> So, there are options.

I prefer to use a GH action for that, preferably using the new Trusted Publisher support instead of personal access tokens.

> Do I need to ask @davehunt for me to be added as a maintainer on PyPI?

Indeed, I just checked and I don't have permissions either.

@davehunt could you give us both access please?

**Comment by okken** [comment]
@nicoddemus Awesome. Thanks.

**Comment by davehunt** [comment]
> @davehunt could you give us both access please?

Sorry, I didn't respond here. Do you still need something from me?

**Comment by okken** [comment]
> > @davehunt could you give us both access please?
> 
> Sorry, I didn't respond here. Do you still need something from me?

Nope. Everything is good.

**Pull request merged**